src/main/java/com/authy/api/Tokens.java

```diff
@@ -26,29 +29,24 @@ public Tokens(String uri, String key, boolean testFlag) {
         super(uri, key, testFlag);
     }
 
-    public Token verify(int userId, String token) {
+    public Token verify(int userId, String token) throws UnsupportedEncodingException, AuthyException {
         return verify(userId, token, null);
     }
 
-    public Token verify(int userId, String token, Map<String, String> options) {
+    public Token verify(int userId, String token, Map<String, String> options) throws AuthyException, UnsupportedEncodingException {
```

Comment:
Not sure if we should change the method signature, that will break existing clients of the lib.

About the `UnsupportedEncodingException` IMO we don't really need the `URLEncoder.encode` that produces it, neither `token` nor `userId ` should contain characters that requiere special encoding.

Reply:
Agree on the UnsupportedEncodingException, I removed the code that caused that exception to be thrown. But I think not having the AuthyException on the signature was actually a bug, we were failing silently and not giving any signal to the user that an error had happened